wrapper/src/main/java/software/amazon/jdbc/Driver.java

```diff
@@ -134,26 +135,8 @@ public Connection connect(final String url, final Properties info) throws SQLExc
     try {
       final String driverUrl = url.replaceFirst(PROTOCOL_PREFIX, "jdbc:");
 
-      java.sql.Driver driver;
-      try {
-        driver = DriverManager.getDriver(driverUrl);
-      } catch (SQLException e) {
-        final List<String> registeredDrivers = Collections.list(DriverManager.getDrivers())
-            .stream()
-            .map(x -> x.getClass().getName())
-            .collect(Collectors.toList());
-        throw new SQLException(
-            Messages.get("Driver.missingDriver", new Object[] {driverUrl, registeredDrivers}), e);
-      }
-
-      if (driver == null) {
-        final List<String> registeredDrivers = Collections.list(DriverManager.getDrivers())
-            .stream()
-            .map(x -> x.getClass().getName())
-            .collect(Collectors.toList());
-        LOGGER.severe(() -> Messages.get("Driver.missingDriver", new Object[] {driverUrl, registeredDrivers}));
-        return null;
-      }
+      TargetDriverHelper helper = new TargetDriverHelper();
+      java.sql.Driver driver = helper.getTargetDriver(driverUrl, props);
```

Comment:
NIT: is `getTargetDriver` the right name for this function?
we first try to find if the driver is registered, then try to register the driver of the dialect

after this line, we either had a driver ready, either registered a new driver, either crashed
would `instantiateDriver` a better fit?

Reply:
This method either return a target driver, or raises an exception. 

What could be alternatives for `getTargetDriver` method name?